File: frontend/src/api/mapa-pulso.service.ts
// Acomodo del mapa 3D del Pulso FEXPO.
// Se guarda como UN documento JSON: el mapa se edita y se guarda entero.
import http from '@/api/http'

export interface MapaPulsoDetalleDto {
  /** JSON con el acomodo; null si nunca se editó (se usa el del código). */
  contenido: string | null
  fechaModificacion?: string
  personalizado: boolean
}

export function obtenerMapa() {
  return http.get<MapaPulsoDetalleDto>('/mapa-pulso').then((r) => r.data)
}

export function guardarMapa(contenido: string) {
  return http.put<MapaPulsoDetalleDto>('/mapa-pulso', { contenido }).then((r) => r.data)
}

/** Borra el acomodo guardado: el mapa vuelve al original del código. */
export function restaurarMapa() {
  return http.delete('/mapa-pulso')
}

File: src/main/java/com/uap/control_tickets/services/interfaces/MapaPulsoService.java
package com.uap.control_tickets.services.interfaces;

import com.uap.control_tickets.dto.mapa.MapaPulsoDetalleDto;
import com.uap.control_tickets.dto.mapa.MapaPulsoDto;

/** Acomodo del mapa 3D del Pulso FEXPO. */
public interface MapaPulsoService {

    /** El acomodo guardado; personalizado=false si nunca se editó. */
    MapaPulsoDetalleDto obtener();

    /** Guarda (o reemplaza) el acomodo. */
    MapaPulsoDetalleDto guardar(MapaPulsoDto dto);

    /** Borra el acomodo guardado: el mapa vuelve al original del código. */
    void restaurar();
}
